src/shell/shell-header.tsx

```diff
@@ -4,26 +4,81 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 import {
 	Container,
 	IconButton,
 	Padding,
 	Responsive,
 	useScreenMode,
-	Catcher,
-	Button
+	Catcher
 } from '@zextras/carbonio-design-system';
-import { times } from 'lodash';
+import { find, size } from 'lodash';
+import styled from 'styled-components';
 import Logo from '../svg/carbonio.svg';
 import { SearchBar } from '../search/search-bar';
 import { CreationButton } from './creation-button';
 import { useAppStore } from '../store/app';
+import { useLoginConfigStore } from '../store/login';
+import { useUserSettings } from '../store/account';
+import { SHELL_APP_ID } from '../constants';
+import { DRPropValues } from '../../types';
+
+const CustomImg = styled.img`
+	height: 2rem;
+`;
 
 const ShellHeader: FC<{
 	mobileNavIsOpen: boolean;
 	onMobileMenuClick: () => void;
 }> = ({ mobileNavIsOpen, onMobileMenuClick, children }) => {
+	const { carbonioWebUiAppLogo, carbonioWebUiDarkAppLogo, carbonioWebUiDarkMode } =
+		useLoginConfigStore();
+	const settings = useUserSettings();
+	const settingReceived = useMemo(
+		() => size(settings.prefs) > 0 || size(settings.attrs) > 0 || size(settings.props) > 0,
+		[settings]
+	);
+	const [darkModeEnabled, setDarkModeEnabled] = useState(false);
+
+	useEffect(() => {
+		if (settingReceived) {
+			const result = find(settings?.props ?? [], {
+				name: 'zappDarkreaderMode',
+				zimlet: SHELL_APP_ID
+			})?._content as DRPropValues;
+			if (result) {
+				setDarkModeEnabled(
+					(result === 'auto' && window.matchMedia('(prefers-color-scheme: dark)').matches) ||
+						result === 'enabled'
+				);
+			} else {
+				setDarkModeEnabled(
+					(carbonioWebUiDarkMode === undefined &&
+						window.matchMedia('(prefers-color-scheme: dark)').matches) ||
+						(carbonioWebUiDarkMode !== undefined && carbonioWebUiDarkMode)
+				);
+			}
+		}
+	}, [settings, carbonioWebUiDarkMode, settingReceived]);
+
+	useEffect(() => {
+		const setCallback = (event: MediaQueryListEvent): void => {
+			setDarkModeEnabled(event.matches);
+		};
+		window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', setCallback);
```

Comment:
it could be convenient to create a util function to access `window.matchMedia('(prefers-color-scheme: dark)')` since it is used a lot and it contains a string which is prone to typos